### ISSUE TITLE
[no-unused-vars] false positive when imported type used inside a function with parameter with the same name

### ISSUE THREAD

**Issue by ShPelles** [typescript-eslint/typescript-eslint#1918]
**Repro**

```JSON
{
    "rules": {
        "no-unused-vars": "off",
        "@typescript-eslint/no-unused-vars": ["error"]
    }
}
```

```TS
import { myType } from './myModule';

function fn(myType: myType) {
    const myVar: myType = 1234;
}
```

**Expected Result**
No linting issue is raised

**Actual Result**
Linter raises an error: 'myType' is defined but never used. eslint(@typescript-eslint/no-unused-vars) [1, 10]

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.28.1-alpha.0` |
| `@typescript-eslint/parser`        | `2.28.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `10.15.3` |
| `npm`                              | `6.4.1` |

**Comment by bradzacher** [comment]
See https://github.com/typescript-eslint/typescript-eslint/issues/1856

Please use the search before filing an issue.
